index.js

```diff
@@ -118,4 +128,31 @@ async function parseTemplate(yamlString, templateFactory) {
     }
 }
 
-module.exports = parseTemplate;
+// eslint-disable-next-line require-jsdoc
+async function parsePipelineTemplate(yamlString) {
+    let configToValidate;
+
+    try {
+        configToValidate = await loadTemplate(yamlString);
+        const config = await validatePipelineTemplateStructure(configToValidate);
+
+        return {
+            errors: [],
+            template: config
+        };
+    } catch (err) {
+        if (!err.details) {
+            throw err;
+        }
+
+        return {
+            errors: err.details,
+            template: configToValidate
+        };
+    }
+}
+
+module.exports = {
+    parseTemplate: parseTemplate(),
+    parsePipelineTemplate: parsePipelineTemplate()
```

Comment:
This will be a backward incompatible change. We should make sure to prefix the PR title (commit messages) with `perf(2135)` prefix to bump up major version
https://github.com/semantic-release/semantic-release#how-does-it-work

```suggestion
    parseJobTemplate: parseJobTemplate,
    parsePipelineTemplate: parsePipelineTemplate
```